sdk/src/accounts/webSocketProgramAccountSubscriber.ts

```diff
@@ -140,12 +146,17 @@ export class WebSocketProgramAccountSubscriber<T>
 	}
 
 	unsubscribe(): Promise<void> {
+		this.isUnsubscribing = true;
+		clearTimeout(this.timeoutId);
+		this.timeoutId = undefined;
+
 		if (this.listenerId) {
 			const promise =
 				this.program.provider.connection.removeAccountChangeListener(
 					this.listenerId
 				);
 			this.listenerId = undefined;
+			this.isUnsubscribing = false;
```

Comment:
Since the promise is returned immediately, `this.isUnsubscribing` can be set to false before the promise actually resolves, is that ok? or do you need to wait for the promise to resolve before clearing listenerId and isSubscribing